lineapy/plugins/pipeline_writers.py

```diff
@@ -302,48 +342,172 @@ def task_iris_pred():
+        variables logic of the artifact fucntion call_block and wrap them into a
+        new function definition.
+        """
+        task_definitions: Dict[str, TaskDefinition] = dict()
+        unused_input_parameters = set(
+            self.artifact_collection.input_parameters
+        )
+        for session_artifacts in self.session_artifacts_sorted:
+            session_input_parameters_spec = (
+                session_artifacts.get_session_input_parameters_spec()
+            )
+            for nc in session_artifacts.artifact_nodecollections:
+                all_input_variables = sorted(list(nc.input_variables))
+                artifact_user_input_variables = [
+                    var
+                    for var in all_input_variables
+                    if var in unused_input_parameters
+                ]
+                user_input_var_typing_block = [
+                    f"{var} = {session_input_parameters_spec[var].value_type}({var})"
+                    for var in artifact_user_input_variables
+                ]
+                unused_input_parameters.difference_update(
+                    set(artifact_user_input_variables)
+                )
+                input_var_loading_block = [
+                    f"{var} = pickle.load(open('/tmp/{self.pipeline_name}/variable_{var}.pickle','rb'))"
+                    for var in all_input_variables
+                    if var not in artifact_user_input_variables
+                ]
+                function_call_block = nc.get_function_call_block(
+                    indentation=0, source_module=f"{self.pipeline_name}_module"
+                )
+                return_var_saving_block = [
+                    f"pickle.dump({var},open('/tmp/{self.pipeline_name}/variable_{var}.pickle','wb'))"
+                    for var in nc.return_variables
+                ]
+                TASK_FUNCTION_TEMPLATE = load_plugin_template(
+                    "task_function.jinja"
+                )
+                function_definition = TASK_FUNCTION_TEMPLATE.render(
+                    function_name=nc.safename,
+                    user_input_variables=", ".join(
+                        artifact_user_input_variables
+                    ),
+                    typing_blocks=user_input_var_typing_block,
+                    loading_blocks=input_var_loading_block,
+                    call_block=function_call_block,
+                    dumping_blocks=return_var_saving_block,
+                    indentation_block=" " * indentation,
+                )
+                task_def: TaskDefinition = {
+                    "definition": function_definition,
+                    "user_input_variables": artifact_user_input_variables,
+                }
+                task_definitions[nc.safename] = task_def
+
+        return task_definitions
+
 
-def get_task_definition(
-    nc: NodeCollection, pipeline_name: str, indentation=4
+def get_session_task_definition(
```

Comment:
Wonder if this is better to be put under `BasePipelineWriter` as a static/class method.